mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -304,14 +306,26 @@ class KMeans @Since("1.5.0") (
   @Since("1.5.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
-  @Since("2.0.0")
+   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
+    /*
+      Before fit the dataset, we check storage level of the rdd
+      if rdd is not cached, we cached our RDD
+    */
```

Comment:
The comment may not be necessary, as it's used elsewhere and clear by the code itself